src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -759,8 +762,10 @@ private bool ParseRawHeaderValues(HeaderDescriptor descriptor, HeaderStoreItemIn
                     }
 
                     // At this point all values are either in info.ParsedValue, info.InvalidValue, or were removed since they
-                    // contain invalid newline chars. Reset RawValue.
+                    // contain invalid newline chars. Reset RawValue, but preserve OriginalRawValue to keep data for non-validating enumeration.
+                    object? originalRawValue = info.OriginalRawValue;
```

Comment:
Why is this retrieving `info.OriginalRawValue` and then just setting it back to the same value?

Reply:
And similarly below

Reply:
It's a leftover from an initial version where `RawValue` setter also updated `OriginalRawValue`. I forgot to remove it, will do now.